### PR TITLE
Add facet search method

### LINKED ISSUE #470: [v1.3] Search in facet values `facetSearch`
⚠️ This issue is generated, it means the examples and the namings do not necessarily correspond to the language of this repository.
Also, if you are a maintainer, please add any clarification and instructions about this issue.

Sorry if this is already wholly/partially implemented. Feel free to let me know about the state of this issue in the repo.

Related to https://github.com/meilisearch/integration-guides/issues/280

----

This issue is divided into two sections, first, you need to make the implementation, and second, you must update the code-samples (no one likes outdated docs, right?).

### New implementation

Related to:
- issue: https://github.com/meilisearch/meilisearch/issues/3611
- discussion: https://github.com/meilisearch/product/discussions/515#discussioncomment-6240970
- spec: https://github.com/meilisearch/specifications/pull/246

Users of bigger datasets could have difficulty interacting with vast lists of facets data since, until today, it was not possible to search on them.

To introduce this feature into the SDKs, we must:

Create a new method called `facetSearch(facetSearchQuery)`/`facet_search(facetSearchQuery)`
Besides all the other parameters that a regular `SearchQuery` can take. The new `FacetSearchQuery` can take a `facetName`, `facetQuery`, `filter`, `q`, `matchingStrategy`.

Params definition:
```yaml
facetName:
  type: string
  required: true
  example: '"genres"'
facetQuery:
  type: string
  example: '"Horror"'
q:
  type: string
  default: '""'
  example: '"Back to the future"'
matchingStrategy:
  type: string
  default: 'last'
filter: // https://www.meilisearch.com/docs/learn/fine_tuning_results/filtering
  type: array
  example: '"groupId = 1 AND genre = adventure"'
```

### Code samples

Inside of this file: `.code-samples.meilisearch.yml`:

1. Create a new entry with this key `facet_search_1` containing a call to the new `facetSearch` method.
Refer to this curl example, and "translate" it by using go methods: https://github.com/meilisearch/documentation/blob/e4fdc3d485c22e31a045b8a71bacf18b6a7d50e0/.code-samples.meilisearch.yaml#L1069.

2. Add new code sample `facet_search_3`
Refer to this curl example, and "translate" it by using go methods: https://github.com/meilisearch/documentation/blob/e4fdc3d485c22e31a045b8a71bacf18b6a7d50e0/.code-samples.meilisearch.yaml#L1087

TODO:
- [ ] Add the new method `facetSearch`
- [ ] Update the code-samples accordingly
- [ ] Add integration tests

### PR DESCRIPTION
**Pull request by migueltarga**
# Pull Request

## Related issue
Fixes #470

## What does this PR do?
- Implements Facet Search introduced on version 1.3 
  Reference: https://www.meilisearch.com/docs/reference/api/facet_search

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: .code-samples.meilisearch.yaml [hunk 1]
```diff
@@ -98,6 +98,17 @@ multi_search_1: |-
       },
     },
   })
+facet_search_1: |-
+  client.Index("movies").FacetSearch(&meilisearch.FacetSearchRequest{
```
Comment (curquiza) [inline .code-samples.meilisearch.yaml:102]:
```suggestion
  client.Index("books").FacetSearch(&meilisearch.FacetSearchRequest{
```

**Review by curquiza — APPROVED** [review]
Thank youuuu

bors merge

**Comment by migueltarga** [comment]
@curquiza Thank you for the approval!

**Comment by curquiza** [comment]
I wait for a few days to see if I can merge https://github.com/meilisearch/meilisearch-go/pull/544 as well and put it in the same release
If I have no answer I will release on Wednesday

**Pull request merged**